webapp-django/crashstats/signature/views.py

```diff
@@ -424,3 +424,112 @@ def signature_graph_data(request, channel):
     )
 
     return data
+
+
+@waffle_switch('signature-report')
+def signature_summary(request):
+    '''Return a list of specific aggregations. '''
+    params = get_validated_params(request)
+    if isinstance(params, http.HttpResponseBadRequest):
+        # There was an error in the form, let's return it.
+        return params
+
+    signature = params['signature'][0]
+
+    data = {}
+
+    current_query = request.GET.copy()
+    if 'page' in current_query:
+        del current_query['page']
+
+    data['params'] = current_query.copy()
+
+    params['signature'] = '=' + signature
+    params['_results_number'] = 0
+    params['_facets'] = [
+        'platform',
+        'cpu_name',
+        'process_type',
+        'flash_version',
+    ]
+    params['_histogram.uptime'] = ['product']
+    params['_histogram_interval.uptime'] = 60
+
+    # If the user has permissions, show exploitability.
+    all_fields = SuperSearchFields().get()
+    if has_permissions(
+        request.user, all_fields['exploitability']['permissions_needed']
+    ):
+        params['_histogram.date'] = ['exploitability']
+
+    api = SuperSearchUnredacted()
+    try:
+        search_results = api.get(**params)
+    except models.BadStatusCodeError, e:
+        # We need to return the error message in some HTML form for jQuery to
+        # pick it up.
+        return http.HttpResponseBadRequest('<ul><li>%s</li></ul>' % e)
```

Comment:
We need to fix that. It's warped and just wrong. 
That idea of showing a bullet point list back to the user is commonly something we do to describe validation errors that are caused by the user. In this case, it's our fault that the squirrels in the machines are on strike.

Perhaps I just don't get the full context and this PR might not be the right place to fix the world but we ought to not forget this and address it at some point soon.